maps-compose/src/main/java/com/google/maps/android/compose/GoogleMap.kt

```diff
@@ -175,6 +175,8 @@ private fun MapLifecycle(mapView: MapView) {
         onDispose {
             lifecycle.removeObserver(mapLifecycleObserver)
             context.unregisterComponentCallbacks(callbacks)
+            mapView.onDestroy()
+            mapView.removeAllViews()
```

Comment:
Is this line necessary? I would think `mapView.onDestroy()` should suffice.

Reply:
I was encountering a google map leak similar to this one: https://stackoverflow.com/questions/65358900/leakcanary-detecting-a-leak-in-my-java-android-app-google-maps and only by removing the views did it get cleaned by GC. It happened when having a route with markers. Do you want me to reproduce locally and add screenshots + leakcanary logs? This is potentially something that could be fixed in google library. There developers can override onDestroy and fix the leak. In this wrapper, they can't.

Reply:
Thanks for clarifying. Let's keep this line here as I don't see any impact as the MapView is being removed/destroyed anyway. I'd like to merge in #193 before merging this in.